src/qt/guiutil.cpp

```diff
@@ -766,10 +766,10 @@ QString NetworkToQString(Network net)
     assert(false);
 }
 
-QString ConnectionTypeToQString(ConnectionType conn_type, bool relay_txes)
+QString ConnectionTypeToQString(ConnectionType conn_type, bool relay_txes, bool relevant_services)
 {
     switch (conn_type) {
-    case ConnectionType::INBOUND: return relay_txes ? QObject::tr("Inbound Full Relay") : QObject::tr("Inbound Block Relay");
+    case ConnectionType::INBOUND: return !relay_txes && relevant_services ? QObject::tr("Inbound Block Relay") : QObject::tr("Inbound");
```

Comment:
This heuristic seems to be working fine. At least I can't find a counter example right now that can be observed on today's network. Though, it still seems fragile to assume that the type of the inbound connection can be deduced based on fuzzy logic.

If it was fine to assume so, then BIP https://github.com/bitcoin/bips/pull/1052 wouldn't be needed.

Reply:
If helpful, the logic is from https://github.com/bitcoin/bitcoin/blob/16b784d953365bb2d7ae65acd2b20a79ef8ba7b6/src/net.cpp#L872 in our inbound eviction logic.

Reply:
WDYT, display "Inbound" for all inbounds or add this info? (-netinfo currently shows it)

Reply:
I think plain "Inbound" makes most sense because it is neither possible to enumerate all inbound connection types, nor is it possible to determine them (if they were enumerable).

Reply:
Yes, it seems we will need to deduce some types of connections from other properties, as currently done to obtain the inbound block relay peers for eviction protection. Take for instance this https://github.com/bitcoin/bitcoin/pull/20726#issuecomment-754802089:

>> #19670 introduced a workaround to deduce inbound block-relay-only peers and protect them from eviction. After this PR, do you think we could revisit to make it more direct?

> I think we would wait until this has been deployed for a while and then (potentially) change the eviction logic. Two things occur to me on this point:
> - Older software will continue to use the old way of (not) communicating their intentions, via the fRelayTxes flag. So to accommodate older software we may not want to switch to keying off this p2p message for a little while at least.
> - It seems most of the feedback has been to shy away from encoding overly broad semantics into a single p2p message; while I think that is fine it also means that from our peer's point of view, they will have to infer the connection "type" from the properties of the peer, rather than from the peer declaring its full intentions in a single message. I think for now we can treat "will never relay txes" as essentially synonymous with these block-relay connections, but perhaps in the future there will be some distinction, and all a peer can do is rely on some set of flags that it might use to preference some peers over others -- which might put us exactly back to where we are today, with us just keying off of whether fRelayTxes is false, anyway.

So while I'd like to see this type of peers easily, yes for now it may be more prudent to just use plain "Inbound" and display other properties useful for inferring the connection status, like fRelayTxes, separately.

Reply:
Even with BIP https://github.com/bitcoin/bips/pull/1052 you couldn't enumerate all possible inbound connection types. This is simply impossible.

Using properties of the connection like service bits or relay flags (and maybe later a `disabletx` message) or other statistics like last sent tx/block to aid eviction can be done without exactly enumerating the connection types.

Reply:
I don't disagree, though as a user I'd like to see this information...either directly as proposed here, or indirectly by manually looking at the `inbound` + `relaytxes` + `services`/`servicesnames` getpeerinfo boolean fields. Anyway, closing for now, replaced by #203.

Reply:
Services and inbound status should be shown in the gui. I don't recall whether version.fRelay is shown

Reply:
Yes, inbound and services are. Proposing now to add frelaytxes and bip152 high bandwidth.